drivers/clock_control/clock_control_mcux_syscon.c

```diff
@@ -623,11 +623,58 @@ static int SYSCON_SET_FUNC_ATTR mcux_lpc_syscon_clock_control_set_subsys_rate(
 	}
 }
 
+static int mcux_lpc_syscon_clock_control_configure(const struct device *dev,
+						   clock_control_subsys_t sub_system, void *data)
+{
+#ifdef CONFIG_SOC_SERIES_RW6XX
+	uint32_t clock_name = (uint32_t)sub_system;
+	int flexcomm_num = -1;
+
+	switch (clock_name) {
+	case MCUX_FLEXCOMM0_CLK:
+	case MCUX_FLEXCOMM0_LP_CLK:
+		flexcomm_num = 0;
+		break;
+	case MCUX_FLEXCOMM1_CLK:
+	case MCUX_FLEXCOMM1_LP_CLK:
+		flexcomm_num = 1;
+		break;
+	case MCUX_FLEXCOMM2_CLK:
+	case MCUX_FLEXCOMM2_LP_CLK:
+		flexcomm_num = 2;
+		break;
+	case MCUX_FLEXCOMM3_CLK:
+	case MCUX_FLEXCOMM3_LP_CLK:
+		flexcomm_num = 3;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	if (flexcomm_num >= 0) {
+		static uint32_t frgclksels[4];
+		static uint32_t frgctls[4];
+
+		if (clock_name & 0x80) {
```

Comment:
what is the special bit at 0x80?

Reply:
It's just decoding the arbitrary value given in the DT clock header that is specific for this driver. I could give the bit a macro name in the header

Reply:
I think I will just make a descriptive macro mask on the next push